teammapper-backend/src/map/utils/tests/mapFactories.ts

```diff
@@ -20,6 +20,17 @@ export const createMmpMap = (overrides = {}): MmpMap => ({
     ...overrides
 })
 
+export const createMmpRootNode = (mapId: string, overrides = {}): Partial<MmpNode> => ({
```

Comment:
Why Partial? Imho a factory should return a valid object?

Reply:
Since this method is only used for particular test cases, I chose Partial to avoid having to specify things like colorsBackground, colorsName etc - these attributes don't add anything to the test except a lot of extra lines of code.

Reply:
ok. what I do not like that much is this part:

```
const rootNode = createMmpRootNode(localMap.id) as MmpNode
```

So you are casting now as you only have a partial but you want to have a full one ;) . Either you merge with something to get the values, or you have to construct a complete object. But this doesn't look like best practice to me.

Reply:
Yeah I'm creating a full node now. The other reason I was keen on using Partial was because I'm not using repositories in the factories (yet). Typescript itself doesn't understand that nodeMap is actually nullable, since that's a NestJS property, so it expects both nodeMap and nodeMapId to be present - same with nodeParent. Tests work fine, but I think `mapFactories.ts` could do with a refactor to use repositories.